pkg/machineoutput/event_logging.go

```diff
@@ -0,0 +1,258 @@
+	json := MachineEventWrapper{
+		ReportError: &ReportError{
+			Error:            errorVal.Error(),
+			AbstractLogEvent: AbstractLogEvent{Timestamp: timestamp},
+		},
+	}
+
+	OutputSuccessUnindented(json)
+}
+
+// GetEntry will return the JSON event parsed from a single line of '-o json' machine readable console output.
+func (w MachineEventWrapper) GetEntry() (MachineEventLogEntry, error) {
+
+	if w.DevFileCommandExecutionBegin != nil {
+		return w.DevFileCommandExecutionBegin, nil
+
+	} else if w.DevFileCommandExecutionComplete != nil {
+		return w.DevFileCommandExecutionComplete, nil
+
+	} else if w.LogText != nil {
+		return w.LogText, nil
+
+	} else if w.ReportError != nil {
+		return w.ReportError, nil
+
+	} else {
+		return nil, errors.New("unexpected machine event log entry")
+	}
+
+}
+
+// GetTimestamp returns the timestamp element for this event.
+func (c AbstractLogEvent) GetTimestamp() string { return c.Timestamp }
+
+// GetType returns the event type for this event.
+func (c DevFileCommandExecutionBegin) GetType() MachineEventLogEntryType {
+	return TypeDevFileCommandExecutionBegin
+}
+
+// GetType returns the event type for this event.
+func (c DevFileCommandExecutionComplete) GetType() MachineEventLogEntryType {
+	return TypeDevFileCommandExecutionComplete
+}
+
+// GetType returns the event type for this event.
+func (c LogText) GetType() MachineEventLogEntryType { return TypeLogText }
+
+// GetType returns the event type for this event.
+func (c ReportError) GetType() MachineEventLogEntryType { return TypeReportError }
+
+// MachineEventLogEntryType indicates the machine-readable event type from an ODO operation
+type MachineEventLogEntryType int
+
+const (
+	// TypeDevFileCommandExecutionBegin is the entry type for that event.
+	TypeDevFileCommandExecutionBegin MachineEventLogEntryType = 0
+	// TypeDevFileCommandExecutionComplete is the entry type for that event.
+	TypeDevFileCommandExecutionComplete MachineEventLogEntryType = 1
+	// TypeLogText is the entry type for that event.
+	TypeLogText MachineEventLogEntryType = 2
+	// TypeReportError is the entry type for that event.
+	TypeReportError MachineEventLogEntryType = 3
+)
+
+// GetCommandName returns a command if the MLE supports that field (otherwise empty string is returned).
+// Currently used for test purposes only.
+func GetCommandName(entry MachineEventLogEntry) string {
+
+	if entry.GetType() == TypeDevFileCommandExecutionBegin {
+		return entry.(*DevFileCommandExecutionBegin).CommandID
+	} else if entry.GetType() == TypeDevFileCommandExecutionComplete {
+		return entry.(*DevFileCommandExecutionComplete).CommandID
+	} else {
+		return ""
+	}
+
+}
+
+// FindNextEntryByType locates the next entry of a given type within a slice. Currently used for test purposes only.
+func FindNextEntryByType(initialIndex int, typeToFind MachineEventLogEntryType, entries []MachineEventLogEntry) (MachineEventLogEntry, int) {
```

Comment:
Since this is only used for test purposes at the moment, should it be moved into `tests/integration`?